gen/src/MPIgen.jl

```diff
@@ -0,0 +1,42 @@
+module MPIgen
+    using Clang.Generators
+    using MPICH_jll
+
+    generate_signatures(dir = MPICH_jll.artifact_dir) = begin
+        @info "Generate MPI bindings for MPICH in $dir"
+
+        out = joinpath(@__DIR__, "..", "out")
```

Comment:
I wonder if this should be saved into the package src directory instead

Reply:
@simonbyrne, what is your preferred directory structure ?

Reply:
I've decided to move the auto-generated file to `src/auto_generated_api.jl` and added a warning at the beginning of the file advising against editing it.